src/main/java/chess/domain/strategy/pawn/PawnStrategy.java

```diff
@@ -0,0 +1,109 @@
+import chess.domain.dto.PositionDto;
+import chess.domain.dto.req.MoveRequest;
+import chess.domain.strategy.PieceStrategy;
+
+import java.util.List;
+
+public class PawnStrategy implements PieceStrategy {
+
+    @Override
+    public void validateDirection(final MoveRequest request) {
+        List<Position> positions = request.getPositions();
+        String movablePieceColor = request.getMovablePieceColor();
+        PositionDto movablePiecePosition = request.getMovablePieceDto();
+        PositionDto targetPosition = request.getTargetPositionDto();
+
+        boolean moveFrontPosition = canMoveFront(positions, movablePieceColor, movablePiecePosition, targetPosition);
+        if (moveFrontPosition) {
+            return;
+        }
+
+        boolean moveDiagonalPosition = canMoveDiagonal(movablePiecePosition, targetPosition);
+        if (moveDiagonalPosition) {
+            return;
+        }
+        throw new IllegalArgumentException("이동할 수 있는 위치가 아닙니다.");
+    }
+
+    private boolean canMoveFront(List<Position> positions, String movablePieceColor, PositionDto movablePiecePosition, PositionDto targetPosition) {
+        if (isNotAlreadyExistFront(positions, targetPosition) && isSameFile(movablePiecePosition, targetPosition)) {
+            if (isWhitePiece(movablePieceColor)) {
+                return getWhitePosition(movablePiecePosition, targetPosition);
+            }
+            return getBlackPosition(movablePiecePosition, targetPosition);
+        }
+        return false;
+    }
+
+    private boolean canMoveDiagonal(PositionDto movablePiecePosition, PositionDto targetPosition) {
+        if (isDiagonal(movablePiecePosition, targetPosition)) {
+            return true;
+        }
+        throw new IllegalArgumentException("이동할 수 있는 위치가 아닙니다.");
+    }
+
+    private boolean isDiagonal(PositionDto movablePiecePosition, PositionDto targetPosition) {
+        int rankDistance = targetPosition.getRank() - movablePiecePosition.getRank();
+        int fileDistance = targetPosition.getFile() - movablePiecePosition.getFile();
+        return Math.abs(rankDistance) == 1 && Math.abs(fileDistance) == 1;
+    }
+
+    private boolean isNotAlreadyExistFront(List<Position> positions, PositionDto targetPosition) {
+        return positions.stream()
+                .anyMatch(position -> !position.isSamePosition(Position.from(targetPosition.getRank(), targetPosition.getFile())));
+    }
+
+    private boolean getWhitePosition(PositionDto movablePiecePosition, PositionDto targetPosition) {
+        validateWhiteMoveBackWard(movablePiecePosition, targetPosition);
+        if (targetPosition.getRank() - movablePiecePosition.getRank() <= 2) {
+            if (isWhiteFirstStep(movablePiecePosition.getRank())) {
+                return true;
+            }
+            throw new IllegalArgumentException("첫번째 칸 일 때만 2칸 이동할 수 있습니다.");
+        }
+        throw new IllegalArgumentException("첫번째 칸 일 때만 2칸 이동할 수 있습니다.");
+    }
```

Comment:
```suggestion
private boolean getWhitePosition(PositionDto movablePiecePosition, PositionDto targetPosition) {
        validateWhiteMoveBackWard(movablePiecePosition, targetPosition);
        if (targetPosition.getRank() - movablePiecePosition.getRank() <= 2) {
            if (isWhiteFirstStep(movablePiecePosition.getRank())) {
                return true;
            }
        }
        throw new IllegalArgumentException("첫번째 칸 일 때만 2칸 이동할 수 있습니다.");
    }
```
다른 케이스도 전부 두 번씩 예외를 던져주셨는데 구분해주신 이유가 있나요?

Reply:
폰전략 리팩터링을 하면서 시간에 쫓겨 꼼꼼히 보지 못했습니다. 현재 리팩터링 완료하였습니다!